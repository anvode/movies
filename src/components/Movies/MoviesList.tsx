import * as React from 'react';
import { useSelector } from 'react-redux';

import { RootState } from '../../redux/root-reducer';
import MoviesItem from './MoviesItem';

import './Movies.scss';

export interface MoviesListProps {}

const MoviesList: React.FC<MoviesListProps> = () => {
    const { movies } = useSelector((state: RootState) => state.movies);

    return <>
        {movies.length > 0 ? (
            movies.map(({id, title}) => (
                <MoviesItem key={id} id={id} title={title}></MoviesItem>
            ))
        ) : (
            <p>No Movies</p>
        )}

    </>;
};

export default MoviesList;