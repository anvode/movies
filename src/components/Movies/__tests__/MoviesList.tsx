import React from 'react';
import Store from '../../../redux/store';
import { render } from '@testing-library/react';
import MoviesList from '../MoviesList';
import {Provider} from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

describe('MoviesList', () => {
    it('renders without crashingt', () => {
        const { asFragment } = render( <Provider store={Store}><BrowserRouter><MoviesList /></BrowserRouter></Provider>);
        expect(asFragment()).toMatchSnapshot();
    });
});