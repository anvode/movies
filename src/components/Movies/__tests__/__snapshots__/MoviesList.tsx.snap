// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList renders without crashingt 1`] = `
<DocumentFragment>
  <p>
    No Movies
  </p>
</DocumentFragment>
`;
